### PR TITLE
Add OSInfo function to user agent

### LINKED ISSUE #34: Shorter Linux tagging
Question: Currently, Linux is being tagged as "Linux x86_64" for the OS part. It would be nice if it was just "Linux". I'll be happy to write up little PR for this, but I just wanted your opinion on it first.

One proposal for a solution would be to check if "Linux" appears in the OS name and then split on any spaces and take the first part. I don't know if this would work for all Linux variants.

### PR DESCRIPTION
**Pull request by dlebech**
This PR adds a new `OSInfo` function that extracts a prettier name and version number from the raw operating system string and returns a `struct` object containing the info. The struct also contains the existing information parsed by the user agent.

I have not added tests for every single user agent string, because the OS name is based on the existing name parsing. I have tried to handle a few edge cases to make the names slightly prettier (e.g. iPhone's user agent seems to be really weird).

This closes #34 and is related to #30.

I am open to suggestions on the functionality here.

### CODE REVIEW

File: operating_systems.go [hunk 1]
```diff
@@ -6,6 +6,25 @@ package user_agent
...
+	Name string
+
+	// Operating system version, e.g. 7 for Windows 7 or 10.8 for Max OS X Mountain Lion
+	Version string
+
+	// Same as ua.Localization()
+	Localization string
+
+	// Same as ua.Platform()
+	Platform string
```
Comment (mssola) [inline operating_systems.go]: I'm not entirely sure why we'd want either `Localization` and `Platform` here. They already have their own getters which are plain and convenient. What do you think ?

File: operating_systems.go [hunk 2]
```diff
@@ -281,3 +300,60 @@ func (p *UserAgent) OS() string {
...
+
+		if strings.Contains(version, "x86") || strings.Contains(version, "i686") {
+			// x86_64 and i868 are not Linux versions but architectures
+			version = ""
+		} else if version == "X" && name == "Mac OS" {
+			// X is not a version for Mac OS.
+			name = name + " " + version
+			version = ""
+		}
+	}
```
Comment (mssola) [inline operating_systems.go]: I think it would be better to put this specific if-else block in another function.

File: operating_systems.go (hunk 1 above)
Comment (dlebech) [inline operating_systems.go]: Yeah sounds good. I'll remove them :)

**Comment by dlebech** [comment]
@mssola I've updated the PR with the requested changes. Let me know what you think.

**Comment by mssola** [comment]
Thanks 👏
